src/core/EdFi.DataManagementService.Core.External/Interface/IDocumentStoreRepository.cs

```diff
@@ -22,6 +22,11 @@ public interface IDocumentStoreRepository
     /// </summary>
     public Task<GetResult> GetDocumentById(IGetRequest getRequest);
 
+    /// <summary>
+    /// Entry point for get document by resource name requests
+    /// </summary>
+    public Task<GetResult> GetDocumentByResourceName(IGetRequest getRequest, int offset, int limit);
```

Comment:
You'll be surprised to hear that you don't need this at all! You are actually implementing the `IQueryHandler` interface that is in the same folder.